src/Arcus.Security.Secrets.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -63,18 +77,43 @@ public async Task<string> Get(string secretName)
                     throw new SecretNotFoundException(secretName, keyVaultErrorException);
                 }
 
+                // TODO: are there other HTTP status codes that we can relate to secret exceptions?
                 throw;
             }
         }
 
         private async Task<KeyVaultClient> GetClientAsync()
         {
-            if (_keyVaultClient == null)
+            await LockCreateKeyVaultClient.WaitAsync();
+
+            try
+            {
+                if (_keyVaultClient == null)
+                {
+                    // TODO: why is this factory not returning an interface (the 'Microsoft.Azure.KeyVault.IKeyVaultClient' interface)?
```

Comment:
I don't think this matters much in this context to be honest

Reply:
See my previous comment(s)

Reply:
Can you point out which one please?

Reply:
Don't really see the issue but I'm open to change. What would be your suggestion? To mock it probably?

Reply:
Will be done after #55